outlines/generate/regex.py

```diff
@@ -39,8 +40,9 @@ def regex(model, regex_str: str, sampler: Sampler = multinomial()):
 
 
 @regex.register(MLXLM)
-def regex_mlxlm(
-    model: MLXLM,
+@regex.register(Transformers)
```

Comment:
the `_unified` dispatchers will become the default dispatcher as a next step. In this PR it's just used by `MLXLM` and `Transformers`